### PR TITLE
improvements on discovery map view

### PR DESCRIPTION
**Pull request by xuezhaokun**
This PR implements the mockup proposed in issue #1361 
Can anyone help me to deploy this PR on some test environment for people to try it out?

### CODE REVIEW

**Comment by mauriel133** [comment]
@xuezhaokun, I've started deployment at https://playground.hydroshare.org/

@pkdash, I will do some review of the front end changes if that's ok.

**Comment by xuezhaokun** [comment]
@mauriel133 Thanks Mauriel! @dtarb Can you just let some GIS developers to test it and give some feedback? @ChristinaB The latest changes have been deployed at https://playground.hydroshare.org/. You can send it to a group of users for testing. Thanks!

**Comment by mauriel133** [comment]
@xuezhaokun, this looks great! Few suggestions:

 

- Give the map container (#discover-map) 
`'border: 1px solid #DDD'. `
This should help frame the map, specially when it's loading and the background is white.
- Move the geo coder panel to the bottom left corner, above the google logo. In mobile devices the current layout makes the geo coder panel overlap the other controls on top.
- I think the coordinates at the top of the map are no longer needed and do not add much value, so we might want to remove them. 
- Change the height of the container with the coverages so it doesn't cut the last row shown.
From this:
image
To this:
image
- Give the geo coder panel:

>     background-color: rgba(255,255,255,0.75);
>     border: 1px solid #DDD;
>     border-radius: 4px;

The transparency helps the usability of the map. Also give the legend container (#discover-map-legend) for consistency as well:
>     border: 1px solid #DDD;

- Make sure the geo coder box has some margin with the edge of the map.

**Comment by xuezhaokun** [comment]
@mauriel133 I made changes on css as you suggested. I did not totally remove the coordinates information. I just set its display to none, since we still need this information as query parameters. Another thing I noticed, if we give the map container (#discover-map) `border: 1px solid #DDD`, the map will not zoom correctly when it is initialized. Do you have any idea on it?

**Comment by ChristinaB** [comment]
I think this looks great!  Super COOL. What are the pending issues that a user group might help resolve?  Could we post a request for Help on the Help page? I could remove the exploding earth and send people to the playground for beta testing? I just need a simple question that we can collect data on their answers - maybe in a Google Form? I propose open user-developer communications be facilitated on https://pages.hydroshare.org/

**Comment by mauriel133** [comment]
@xuezhaokun, I will do some testing and see if I can find a fix for the zoom problem while framing the map.

I propose the following CSS rules to fix some layout problems on mobile devices as you can see in this capture:
image

```
#geocoder-panel {
    margin-left: 10px;
    width: 300px;
    margin-left: 10px;
    line-height: 30px;
    font-size: 16px;
}
```
```
#geocoder-address {
    width: 210px;
}
```

Change the caption on the zoom button to " Go"

After these changes it should look like this:
image

**Comment by dtarb** [comment]
@ChristinaB see #1361.  There is some ongoing discussion of this design there.  Would the users you have in mind be comfortable joining github and participating in the discussion in github issues.  This may be better than pages.hydroshare.org as anyone can write comments.  A couple of simple questions might be.
1. Compare the discover page deployments at www.hydroshare.org and playground.hydroshare.org and comment on which map interface you find more effective for identifying resources in a geographic location you are interested in.
2. Where resources have geographic coverage that span a box, and there is a potential for overlap (either exactly or intersecting) do they have ideas about how coverages should be displayed and resources identified on the map.

If you are going to pursue this information gathering with users, please coordinate with Zhaokun to make sure playground persists over the time you will be having users look at it.  You also need to make sure that users understand that nothing on playground is permanent, so not to put any important resources there.

I redirected you to the issue #1361, not the pull request as the pull request is cluttered by unit test results and code comparison discussions, and an issue seems better for a functionality discussion.

**Comment by xuezhaokun** [comment]
@mauriel133 It seems that zoom issue only happens on my virtualbox firefox browser. I did not see this problem on playground. I made css modifications as you suggested. I agree that the map is more important than the coverage items, but I think it might be more convenient for users to discover coverage resources from the top. I'd like to keep the coverage items on the top for now. I will mention this suggestion in my next demo to see whether people think it is better to move it to the bottom.

File: hs_core/discovery_form.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
 from django import forms
 
 class DiscoveryForm(FacetedSearchForm):
-    NElat = forms.CharField(label='NE latitude', required=False)
-    NElng = forms.CharField(label='NE longitude', required=False)
-    SWlat = forms.CharField(label='SW latitude', required=False)
-    SWlng = forms.CharField(label='SW longitude', required=False)
-    start_date = forms.DateField(label='From', required=False)
-    end_date = forms.DateField(label='To', required=False)
+    NElat = forms.CharField(widget = forms.HiddenInput(), required=False)
```
Comment (jbradberry) [inline hs_core/discovery_form.py]: Python files touched should be removed from the setup.cfg file.

**Comment by pkdash** [comment]
@xuezhaokun What's the status of this PR? Are you waiting for code review?

**Comment by xuezhaokun** [comment]
@pkdash I am still working on this PR. I am implementing the new design discussed in issue #1361 .

**Comment by pkdash** [comment]
@xuezhaokun Is this ready for review?

**Comment by pkdash** [comment]
@xuezhaokun Is this ready for review?

**Comment by xuezhaokun** [comment]
@pkdash I think it should be ready for another round review.

**Comment by pkdash** [comment]
@mauriel133 Can you review this?

**Comment by pkdash** [comment]
@mauriel133 This needs to be reviewed soon. Kind of high priority.

**Comment by mjstealey** [comment]
@pkdash, @mauriel133 - This is the last PR for release 1.7.6

Can this be reviewed for deployment to beta by COB today? 

Would prefer to hold off on (and I mean I will forcibly revert) any other merges to the develop branch other than this one prior to the release of 1.7.6

**Comment by mauriel133** [comment]
The soonest I can review it is today at around 1 PM MT

**Comment by mauriel133** [comment]
1 - There seems to be an issue where the map takes too long to load and all the user sees is this:

image

Also, we should make the map start loading once the page is loaded and not when the tab is clicked in order to save time.

2 - These icons and labels should be aligned like in a table:
image
Change 'number of sites' to 'Number of Sites" for consistency

Other than that, the code looks good to me. If possible, I suggest to work out point 1 above before this is merged. If not we could potentially work these in the future and have the PR move on. 

+1 given the observation above.

File: hs_core/templates/search/search.html [hunk 2]
```diff
@@ -445,39 +445,51 @@ <h4 class="panel-title">
...
-                            {% endfor %}
-                            <div class="col-sm-12 height-fix">
-                                <div class="text-muted">WGS 84 decimal degrees</div>
 
-                                <a id="manual-zoom-btn" class="btn btn-default" onclick="manualZoomMap()">
-                                    <span class="glyphicon glyphicon-zoom-in"></span> Zoom In
-                                </a>
+                <div id="map-view" class="tab-pane fade">
+                    <div class="row">
+                        <div class="col-xs-12">
```
Comment (mauriel133) [inline hs_core/templates/search/search.html]: Is this content still used?

**Pull request merged**